boards/arm/mimxrt1024_evk/Kconfig.defconfig

```diff
@@ -12,9 +12,6 @@ choice CODE_LOCATION
 	default CODE_FLEXSPI
 endchoice
 
-config FLASH_MCUX_FLEXSPI_NOR
-	default y if FLASH
-
 config FLASH_MCUX_FLEXSPI_XIP
 	default y if FLASH
```

Comment:
for currently unknown reason, zephyr/main doesn't show this as selected, but with this PR things seems to behave as one would expect, causing this `default y` to manifest itself when `MEMC_MCUX_FLEXSPI=n`.

So to avoid strange behavior, you should add:
```suggestion
	default y if FLASH
	depends on MEMC_MCUX_FLEXSPI
```

else you'll end up with `FLASH_MCUX_FLEXSPI_XIP` being selected when the optional choice `FLEXSPI_FLASH_DRIVER` is disabled, as can be seen:
![image](https://user-images.githubusercontent.com/9802583/134736976-28da7d07-4023-49fd-bbd0-15db63e69da6.png)

Reply:
Thank you. Makes sense, I will take a look at this.